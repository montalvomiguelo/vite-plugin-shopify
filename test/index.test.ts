import { describe, expect, it } from 'vitest'
import { build } from 'vite'
import shopify from '../src/index'
import fs from 'fs/promises'

describe('vite-plugin-shopify', () => {
  it('runs as expected', async () => {
    await build({
      plugins: [
        shopify({
          input: [
            'test/__fixtures__/resources/css/theme.css',
            'test/__fixtures__/resources/js/theme.js'
          ],
          themeRoot: 'test/__fixtures__'
        })
      ]
    })

    const tagsHtml = await fs.readFile('test/__fixtures__/snippets/vite-tag.liquid', { encoding: 'utf8' })
    const clientHtml = await fs.readFile('test/__fixtures__/snippets/vite-client.liquid', { encoding: 'utf8' })

    expect(tagsHtml).toMatchSnapshot()
    expect(clientHtml).toMatchSnapshot()
  })
})
