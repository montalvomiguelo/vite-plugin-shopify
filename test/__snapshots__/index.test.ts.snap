// Vitest Snapshot v1

exports[`vite-plugin-shopify > runs as expected 1`] = `
"{%- if vite-tag == 'test/__fixtures__/resources/css/theme.css' -%}
  <link rel=\\"stylesheet\\" href=\\"{{ 'theme-1a84b608.css' | asset_url }}\\">
{%- endif -%}

{%- if vite-tag == 'test/__fixtures__/resources/js/theme.js' -%}
  <script src=\\"{{ 'theme-5455b76e.js' | asset_url }}\\" type=\\"module\\" crossorigin=\\"anonymous\\"></script>
{%- endif -%}"
`;

exports[`vite-plugin-shopify > runs as expected 2`] = `""`;
